src/lib/avoidance/ObstacleAvoidance.hpp

```diff
@@ -1,152 +0,0 @@
-/****************************************************************************
- *
- *   Copyright (c) 2019 PX4 Development Team. All rights reserved.
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions
- * are met:
- *
- * 1. Redistributions of source code must retain the above copyright
- *    notice, this list of conditions and the following disclaimer.
- * 2. Redistributions in binary form must reproduce the above copyright
- *    notice, this list of conditions and the following disclaimer in
- *    the documentation and/or other materials provided with the
- *    distribution.
- * 3. Neither the name PX4 nor the names of its contributors may be
- *    used to endorse or promote products derived from this software
- *    without specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
- * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
- * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
- * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
- * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
- * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
- * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
- * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
- * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
- * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
- * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
- *
- ****************************************************************************/
-
-/**
- * @file ObstacleAvoidance.hpp
- * This class is used to inject the setpoints of an obstacle avoidance system
- * into the FlightTasks
- *
- * @author Martina Rivizzigno
- */
-
-#pragma once
-
-#include <px4_platform_common/defines.h>
-#include <px4_platform_common/module_params.h>
-#include <commander/px4_custom_mode.h>
-#include <drivers/drv_hrt.h>
-
-#include <uORB/Publication.hpp>
-#include <uORB/Subscription.hpp>
-#include <uORB/topics/position_controller_status.h>
-#include <uORB/topics/vehicle_command.h>
-#include <uORB/topics/vehicle_status.h>
-#include <uORB/topics/vehicle_trajectory_bezier.h>
-#include <uORB/topics/vehicle_trajectory_waypoint.h>
```

Comment:
I believe we could even remove these two uORB messages and sub-messages. They were only used for avoidance - @bastianhjaeger would you agree?
- [VehicleTrajectoryBezier.msg](https://github.com/PX4/PX4-Autopilot/blob/main/msg/VehicleTrajectoryBezier.msg)
   - Usage: https://github.com/search?q=repo%3APX4%2FPX4-Autopilot%20vehicle_trajectory_bezier&type=code
- [VehicleTrajectoryWaypoint.msg](https://github.com/PX4/PX4-Autopilot/blob/main/msg/VehicleTrajectoryWaypoint.msg)
   - Usage: https://github.com/search?q=repo%3APX4%2FPX4-Autopilot%20vehicle_trajectory_waypoint&type=code
- [TrajectoryBezier.msg](https://github.com/PX4/PX4-Autopilot/blob/main/msg/TrajectoryBezier.msg)
- [TrajectoryWaypoint.msg](https://github.com/PX4/PX4-Autopilot/blob/main/msg/TrajectoryWaypoint.msg)

Reply:
Looking deeper still, we could probably then also remove the entire Bezier library.
https://github.com/PX4/PX4-Autopilot/tree/main/src/lib/bezier

Reply:
`Bezier` can for sure go. If `vehicle_trajectory_waypoint` is not used nowhere, than sure too. Good catch.

Reply:
Thanks for the hint, I removed all these messages now.

Reply:
That doubles the flash savings to 10k